dockerfiles/theia/README.md

```diff
@@ -1,33 +1,97 @@
 # How to Build Theia Image
 
 ## Build image manually using build scripts
-To build Theia image manually You could use build.sh scripts.
-Dockerfile uses multi-stage build feature, so first of all You need build builder image -
-theia-dev:
 
-Example:
+To build all the images you can easily run [build.sh](../../build.sh) script located in the repository root. It will build container with Theia editor, dev container with tools to develop on TypeScript and containers containing remote plugins. 
 
-```shell
-$ ../theia-dev/build.sh
+```bash
+./build.sh
+```
+
+CI for checking any pull request in this repository runs this script with `--pr` directive
+```bash
+`build.sh --pr`.
+```
+
+**Note**: `--pr` will build only [limited set](../../docker_image_build.include) of docker images.	
+
+## How to build own Che Theia image with Docker
+
+[Che-Theia](Dockerfile) image is based on [Theia-Dev](../theia-dev/Dockerfile) image, which contains a set of tools for TypeScript development.
+Also theia-dev image includes `@theia/generator-plugin` which can be easily used in both Theia-Dev and Che-Theia containers.
+
+Build script always use current Che-Theia sources. To test your changes you don't need to push your changes somewhere. Just go to `dockerfiles/theia` directory and run:
+
+```bash	
+./build.sh --build-args:${GITHUB_TOKEN},THEIA_VERSION=master --tag:next --branch:master --git-ref:refs\\/heads\\/master
 ```
 
-After that, You can build Theia image.
+Where [`${GITHUB_TOKEN}`](#github-token) is your GitHub API token, it's used for fetching some vscode libraries that placed on GitHub releases. Without that token build will fail.
+
+This script will build new docker image `eclipse/che-theia:next`. The image will contain Che-Theia based on Theia from master branch.
+
+## Build arguments
 
-Example:
+General command to build Che-Theia with all possible argiments:
 
 ```bash
-$ ./build.sh --build-args:GITHUB_TOKEN=$GITHUB_TOKEN,THEIA_VERSION=0.3.18 --git-ref:refs\\/heads\\/master
+./build.sh --build-args:${GITHUB_TOKEN},THEIA_VERSION=${THEIA_VERSION} --tag:${IMAGE_TAG} --branch:${THEIA_BRANCH} --git-ref:${THEIA_GIT_REFS} --skip-tests
 ```
 
+### GitHub Token
+
+**`${GITHUB_TOKEN}`**
+
+Some Theia dependencies placed on GitHub repository and fetched from the GitHub by using GitHub API. For that you have to generate [new GitHub API token](https://github.com/settings/tokens) and pass it to the script to have proper build.
+
+Parameter `${GITHUB_TOKEN}` is mandatory.
+
+
+## Theia Branch
+
+**`${THEIA_BRANCH}`**
+
+It specifies tag or branch for base Theia. Build script clones Theia sources, switches to that branch/tag, and only then clones Che-Theia.
+
+Parameter `${THEIA_BRANCH}` is optional. If it's not specified, the default value `'master'` will be used.
+
+CI uses 
+- `master` branch to generate `eclipse/che-theia:next`
```

Comment:
maybe add link to the job in Jenkins that is using this and that

Reply:
Done